File: Java/src/main/java/com/zooz/extended/java/lib/ZooZQueryTypes/StatusAndDatesQuery.java
package com.zooz.extended.java.lib.ZooZQueryTypes;

import com.zooz.extended.java.lib.model.NVPs;
import com.zooz.extended.java.lib.model.QueryType;
import com.zooz.extended.java.lib.model.enums.Queries;

import java.util.Date;

/**
 * Get all the transactions details between specific dates followed by a specific status
 */
public class StatusAndDatesQuery extends QueryType {
    private String paymentStatus;
    private String startDate, endDate;

    /**
     * @param paymentStatus:    can be one of the following: Pending, TPCPending, AuthorizationProcess, Succeed, Failed,
     *                          UserReject, POSCanceled , Timeout
     * @param startDate:        The earliest date for the requested transactions
     * @param endDate:          The latest date for the requested transactions
     */
    public StatusAndDatesQuery(String paymentStatus, String startDate, String endDate) {
        this.paymentStatus = paymentStatus;
        this.startDate = startDate;
        this.endDate = endDate;
    }

    @Override
    protected void addQueryParametersToNVP(NVPs nvps) {
        nvps.add("paymentStatus", paymentStatus);
        addDatesToNVP(nvps, startDate, endDate);
    }

    @Override
    protected Queries getQuery() {
        return Queries.statusAndDates;
    }
}

File: C#/ExtendedServerAPI/StatusAndDatesQuery.cs
using AbstractClasses;
using ZooZEnums;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ZooZQueryTypes
{
    /// <summary>
    /// Get all the transactions details between specific dates followed by a specific status  </summary>
    class StatusAndDatesQuery : QueryType
    {
        private string paymentStatus { get; set; }
        private DateTime? startDate { get; set; }
        private DateTime? endDate { get; set; }

        /// <summary>
        /// Class constructor.
        /// </summary>
        /// <param name="paymentStatus">
        /// Can be one of the following:
        /// Pending, TPCPending, AuthorizationProcess, Succeed,	Failed, UserRejected, POSCanceled, Timeout </param>
        /// <param name="startDate">
        /// The earliest date for the requested transactions. </param>
        /// <param name="endDate">
        /// The latest date for the requested transactions. </param>
        public StatusAndDatesQuery(string paymentStatus, DateTime? startDate, DateTime? endDate)
            : base()
        {
            this.paymentStatus = paymentStatus;
            this.startDate = startDate;
            this.endDate = endDate;
        }

        protected override void addQueryParametersToKvpList(List<KeyValuePair<string, object>> kvpList)
        {
            kvpList.Add(new KeyValuePair<string, Object>("&paymentStatus=", paymentStatus));
            addDatesToKvp(kvpList, startDate, endDate);
        }

        protected override TypeOfQuery getQueryType()
        {
            return TypeOfQuery.statusAndDates;
        }


    }
}
